tests/tests/tier0/bluechi-is-online-system-wait/test_bluechi_is_online_system_wait.py

```diff
@@ -0,0 +1,205 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.bluechi_is_online import BluechiIsOnline
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+from bluechi_test.util import Timeout, get_test_env_value_int
+
+LOGGER = logging.getLogger(__name__)
+
+AGENT_ONE = "agent-one"
+AGENT_TWO = "agent-two"
+IMMEDIATE_RETURN_TIMEOUT_MS = get_test_env_value_int("IMMEDIATE_RETURN_TIMEOUT", 1000)
+WAIT_PARAM_VALUE_MS = get_test_env_value_int("WAIT_PARAM_VALUE", 5000)
+SLEEP_DURATION = get_test_env_value_int("SLEEP_DURATION", 2)
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_system(
+    bluechi_is_online: BluechiIsOnline,
+    wait_time: int,
+    future: ResultFuture,
+):
+    future.result = bluechi_is_online.system_is_online(
+        wait_time,
+    )
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    # Test 1: Both agents and controller are online
+    LOGGER.debug("Testing with both agents and controller online.")
+    system_result_test_one = ResultFuture()
+    system_thread_wait = threading.Thread(
+        target=check_system,
+        args=(
+            ctrl.bluechi_is_online,
+            IMMEDIATE_RETURN_TIMEOUT_MS,
+            system_result_test_one,
+        ),
+    )
+    system_thread_wait.start()
+    try:
+        with Timeout(SLEEP_DURATION, "Timeout while waiting for system."):
+            system_thread_wait.join()
+    except TimeoutError:
+        LOGGER.debug(
+            "Timeout reached while waiting for system. Attempting to terminate."
+        )
+    assert (
+        system_result_test_one.result
+    ), "System wait command should not produce output when system is running."
```

Comment:
When the system is online (both agents in this case), then `bluechi-is-online system --wait=<x>` should return immediately - so the `with Timeout` is not necessary here:

```suggestion
    system_thread_wait.start()
    system_thread_wait.join()
    
    assert (
        system_result_test_one.result
    ), "System wait command should not produce output when system is running."
```

Reply:
Fixed.